andela/wg-ravens
# Ft api create users

#### What does this PR do?
Enables creation of user accounts through ReST API

#### Description of Task to be completed?
- It should be possible to add new users via the API. Third party applications may want to do this
- They should get a special right from Wger Admin in order to do so
- There should be a way to protect against being flooded with user accounts from third party applications.

#### How should this be manually tested?
- Clone and setup the application locally and have it running
- Login as admin and create a new Gym, add member with `username` as `john_doe` and grant them `gym administrator` role, then note the given `password`
- Get to the command line and execute `python manage.py add_user_rest_api john_doe --enable=true --accounts-limit=5`
- Log out of Wger and login as john_doe and acquire an API Key
- You can now use the API key in the Authorizaiton Header as `Token theGrantedAPIkey` using a tool like Postman to send user data in JSON format including at least `username` and `password`. (**see screenshot for full format**). The endpoint is `http://127.0.0.1:8000/api/v2/user/`
- The user should be created and visible Gym of `john_doe`

#### Any background context you want to provide?
Users may not be able to register more than the allowed accounts per minute as set by the administrator when enabling with the command `add_user_rest_api`. This is done with the `--accounts-limit` option.

#### What are the relevant pivotal tracker stories?
#157589642

#### Screenshots 
<img width="754" alt="screen shot 2018-06-15 at 2 13 30 pm" src="https://user-images.githubusercontent.com/23254512/41465381-8618351c-70a6-11e8-93da-94d374b0731c.png">

<img width="753" alt="screen shot 2018-06-15 at 2 23 32 pm" src="https://user-images.githubusercontent.com/23254512/41465679-c9c7dfa0-70a7-11e8-8783-b76c4b50b6be.png">

<img width="654" alt="screen shot 2018-06-15 at 2 18 26 pm" src="https://user-images.githubusercontent.com/23254512/41465497-09872782-70a7-11e8-9134-22e719162fd9.png">
